diff --git "a/~bansoukou_unpatched\\BiomesOPlenty-1.12.2-7.0.1.2445-universal\\assets\\biomesoplenty\\advancements\\recipes\\buildingBlocks\\mud_bricks.json" "b/bansoukou\\BiomesOPlenty-1.12.2-7.0.1.2445-universal\\assets\\biomesoplenty\\advancements\\recipes\\buildingBlocks\\mud_bricks.json"
index 7cce2d08..d3edc6fb 100644
--- "a/~bansoukou_unpatched\\BiomesOPlenty-1.12.2-7.0.1.2445-universal\\assets\\biomesoplenty\\advancements\\recipes\\buildingBlocks\\mud_bricks.json"
+++ "b/bansoukou\\BiomesOPlenty-1.12.2-7.0.1.2445-universal\\assets\\biomesoplenty\\advancements\\recipes\\buildingBlocks\\mud_bricks.json"
@@ -1,9 +1,4 @@
 {
-  "rewards": {
-    "recipes": [
-      "biomesoplenty:mud_bricks"
-    ]
-  },
   "criteria": {
     "has_log": {
       "trigger": "minecraft:inventory_changed",
@@ -14,18 +9,6 @@
           }
         ]
       }
-    },
-    "has_the_recipe": {
-      "trigger": "minecraft:recipe_unlocked",
-      "conditions": {
-        "recipe": "biomesoplenty:mud_bricks"
-      }
     }
-  },
-  "requirements": [
-    [
-      "has_log",
-      "has_the_recipe"
-    ]
-  ]
+  }
 }
\ No newline at end of file
