diff --git "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\tool\\EntityMiningLaser.java" "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\tool\\EntityMiningLaser.java"
index 510feb71..398471dc 100644
--- "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\tool\\EntityMiningLaser.java"
+++ "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\tool\\EntityMiningLaser.java"
@@ -196,7 +196,7 @@ lbl36:
 
     private void explode() {
         World world = this.func_130014_f_();
-        LaserEvent.LaserExplodesEvent event = new LaserEvent.LaserExplodesEvent(world, (Entity)this, this.owner, this.range, this.power, this.blockBreaks, this.explosive, this.smelt, 5.0f, 0.85f, 0.55f);
+        LaserEvent.LaserExplodesEvent event = new LaserEvent.LaserExplodesEvent(world, (Entity)this, this.owner, this.range, this.power, this.blockBreaks, this.explosive, this.smelt, 3.0f, 0.85f, 0.0f);
         if (MinecraftForge.EVENT_BUS.post((Event)event)) {
             this.func_70106_y();
             return;
