diff --git "a/~bansoukou_unpatched\\TravelersBackpack-1.12.2-1.0.35\\com\\tiviacz\\travelersbackpack\\util\\BackpackUtils.java" "b/bansoukou\\TravelersBackpack-1.12.2-1.0.35\\com\\tiviacz\\travelersbackpack\\util\\BackpackUtils.java"
index d7850fba..664b855b 100644
--- "a/~bansoukou_unpatched\\TravelersBackpack-1.12.2-1.0.35\\com\\tiviacz\\travelersbackpack\\util\\BackpackUtils.java"
+++ "b/bansoukou\\TravelersBackpack-1.12.2-1.0.35\\com\\tiviacz\\travelersbackpack\\util\\BackpackUtils.java"
@@ -7,6 +7,8 @@
  *  com.tiviacz.travelersbackpack.handlers.ConfigHandler
  *  com.tiviacz.travelersbackpack.init.ModBlocks
  *  com.tiviacz.travelersbackpack.tileentity.TileEntityTravelersBackpack
+ *  com.tiviacz.travelersbackpack.util.BackpackUtils$1
+ *  com.tiviacz.travelersbackpack.util.BackpackUtils$1Checker
  *  com.tiviacz.travelersbackpack.util.Reference
  *  net.minecraft.block.Block
  *  net.minecraft.entity.Entity
@@ -30,6 +32,7 @@ import com.tiviacz.travelersbackpack.capability.ITravelersBackpack;
 import com.tiviacz.travelersbackpack.handlers.ConfigHandler;
 import com.tiviacz.travelersbackpack.init.ModBlocks;
 import com.tiviacz.travelersbackpack.tileentity.TileEntityTravelersBackpack;
+import com.tiviacz.travelersbackpack.util.BackpackUtils;
 import com.tiviacz.travelersbackpack.util.Reference;
 import net.minecraft.block.Block;
 import net.minecraft.entity.Entity;
@@ -82,13 +85,51 @@ public class BackpackUtils {
     }

     private static boolean tryPlace(World world, EntityPlayer player, ItemStack itemstack) {
-        int[] positions;
-        int X = (int)player.field_70165_t;
-        int Z = (int)player.field_70161_v;
-        for (int Y : positions = new int[]{0, -1, 1, -2, 2, -3, 3, -4, 4, -5, 5, -6, 6}) {
-            BlockPos spawn = BackpackUtils.getNearestEmptyChunkCoordinatesSpiral((IBlockAccess)world, X, Z, new BlockPos(X, (int)player.field_70163_u + Y, Z), 12, true, 1, (byte)0, false);
-            if (spawn == null) continue;
-            return BackpackUtils.placeBackpack(itemstack, player, world, spawn.func_177958_n(), spawn.func_177956_o(), spawn.func_177952_p(), EnumFacing.UP);
+        int search_height = 128;
+        int px = (int)player.field_70165_t;
+        int pz = (int)player.field_70161_v;
+        int py_candidate = (int)player.field_70163_u;
+        if (py_candidate < 0) {
+            py_candidate = 64;
+        }
+        if (py_candidate > 255) {
+            py_candidate = 191;
+        }
+        int py = py_candidate;
+        int x = 0;
+        int y = 0;
+        int z = 0;
+        BlockPos origin = new BlockPos((double)px, player.field_70163_u, (double)pz);
+        BlockPos[] best = new BlockPos[]{null};
+        Checker C = new Checker(px, py, pz, world, best, origin);
+        int sign_y = 1;
+        int max_steps = 32;
+        block0: while (y < 128) {
+            if (y + py > 0 && y + py < 255) {
+                x = 0;
+                z = 0;
+                int turn = 1;
+                int sign = 1;
+                for (int steps = 0; steps < max_steps; ++steps) {
+                    int i;
+                    for (i = 0; i < turn; ++i) {
+                        if (C.viable(x += sign, y, z)) break block0;
+                    }
+                    for (i = 0; i < turn; ++i) {
+                        if (C.viable(x, y, z += sign)) break block0;
+                    }
+                    ++turn;
+                    sign *= -1;
+                }
+            }
+            if (sign_y == 1) {
+                y += sign_y;
+            }
+            y *= -1;
+            sign_y *= -1;
+        }
+        if (best[0] != null) {
+            return BackpackUtils.placeBackpack(itemstack, player, world, best[0].func_177958_n(), best[0].func_177956_o(), best[0].func_177952_p(), EnumFacing.UP);
         }
         return false;
     }
@@ -101,28 +142,28 @@ public class BackpackUtils {
         if ((block = ModBlocks.TRAVELERS_BACKPACK).func_176198_a(world, new BlockPos(x, y, z), facing)) {
             BlockPos targetPos;
             if (world.func_180495_p(new BlockPos(x, y, z)).func_185904_a().func_76220_a()) {
-                switch (facing) {
-                    case DOWN: {
+                switch (1.$SwitchMap$net$minecraft$util$EnumFacing[facing.ordinal()]) {
+                    case 1: {
                         --y;
                         break;
                     }
-                    case UP: {
+                    case 2: {
                         ++y;
                         break;
                     }
-                    case NORTH: {
+                    case 3: {
                         --z;
                         break;
                     }
-                    case SOUTH: {
+                    case 4: {
                         ++z;
                         break;
                     }
-                    case WEST: {
+                    case 5: {
                         --x;
                         break;
                     }
-                    case EAST: {
+                    case 6: {
                         ++x;
                     }
                 }
@@ -257,4 +298,8 @@ public class BackpackUtils {
         }
         return false;
     }
+
+    static /* synthetic */ boolean access$000(BlockPos x0, BlockPos x1) {
+        return BackpackUtils.areCoordinatesTheSame(x0, x1);
+    }
 }
