diff --git "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFE.java" "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFE.java"
index 06e25614..bdad27ae 100644
--- "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFE.java"
+++ "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFE.java"
@@ -22,6 +22,6 @@ extends TileEntityElectricBlock {
     }
 
     public TileEntityElectricMFE() {
-        super(3, 512, 4000000);
+        super(3, 512, 8000000);
     }
 }
