diff --git "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFSU.java" "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFSU.java"
index a0542158..ac25ce0c 100644
--- "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFSU.java"
+++ "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\block\\wiring\\TileEntityElectricMFSU.java"
@@ -22,6 +22,6 @@ extends TileEntityElectricBlock {
     }
 
     public TileEntityElectricMFSU() {
-        super(4, 2048, 40000000);
+        super(4, 2048, 80000000);
     }
 }
