diff --git "a/~bansoukou_unpatched\\botaniatweaks-1.9.1\\quaternary\\botaniatweaks\\modules\\jei\\RecipeWrapperAgglomeration.java" "b/bansoukou\\botaniatweaks-1.9.1\\quaternary\\botaniatweaks\\modules\\jei\\RecipeWrapperAgglomeration.java"
index 7cf2878f..6f46f7d7 100644
--- "a/~bansoukou_unpatched\\botaniatweaks-1.9.1\\quaternary\\botaniatweaks\\modules\\jei\\RecipeWrapperAgglomeration.java"
+++ "b/bansoukou\\botaniatweaks-1.9.1\\quaternary\\botaniatweaks\\modules\\jei\\RecipeWrapperAgglomeration.java"
@@ -99,4 +99,11 @@ implements IRecipeWrapper {
         }
         GlStateManager.func_179118_c();
     }
+
+    public List<String> getTooltipStrings(int mouseX, int mouseY) {
+        if (mouseX > 35 && mouseX < 140 && mouseY > 60 && mouseY < 65) {
+            return ImmutableList.of((Object)new StringBuffer().append(this.manaCost).append(" mana").toString());
+        }
+        return ImmutableList.of();
+    }
 }
