diff --git "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\ItemBatteryChargeHotbar.java" "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\ItemBatteryChargeHotbar.java"
index 566ecb90..f2cd1499 100644
--- "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\ItemBatteryChargeHotbar.java"
+++ "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\item\\ItemBatteryChargeHotbar.java"
@@ -67,6 +67,8 @@ extends ItemBattery
 implements IBoxable {
     public ItemBatteryChargeHotbar(ItemName name, double maxCharge, double transferLimit, int tier) {
         super(name, maxCharge, transferLimit, tier);
+        this.maxCharge = maxCharge * 2.0;
+        this.transferLimit = transferLimit * 10.0;
     }
 
     @SideOnly(value=Side.CLIENT)
