diff --git "a/~bansoukou_unpatched\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\tile\\TileEntityDisplayStand.java" "b/bansoukou\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\tile\\TileEntityDisplayStand.java"
index 7c4d534c..bdede8cc 100644
--- "a/~bansoukou_unpatched\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\tile\\TileEntityDisplayStand.java"
+++ "b/bansoukou\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\tile\\TileEntityDisplayStand.java"
@@ -37,7 +37,7 @@ import net.minecraftforge.energy.IEnergyStorage;
 public class TileEntityDisplayStand
 extends TileEntityInventoryBase
 implements IEnergyDisplay {
-    public final CustomEnergyStorage storage = new CustomEnergyStorage(80000, 1000, 0);
+    public final CustomEnergyStorage storage = new CustomEnergyStorage(320000, 2000, 0);
     private int oldEnergy;
 
     public TileEntityDisplayStand() {
