diff --git "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\crop\\TileEntityCrop.java" "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\crop\\TileEntityCrop.java"
index 744c6ad1..b0994d2b 100644
--- "a/~bansoukou_unpatched\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\crop\\TileEntityCrop.java"
+++ "b/bansoukou\\industrialcraft-2-2.8.222-ex112\\ic2\\core\\crop\\TileEntityCrop.java"
@@ -305,7 +305,7 @@ implements ICropTile {
             if (this.crop == null) {
                 return;
             }
-            if (this.growthPoints >= this.crop.getGrowthDuration((ICropTile)this)) {
+            if (this.growthPoints >= this.crop.getGrowthDuration((ICropTile)this) / 10) {
                 this.growthPoints = 0;
                 this.setCurrentSize(this.getCurrentSize() + 1);
                 this.dirty = true;
