diff --git "a/~bansoukou_unpatched\\Thaumcraft-1.12.2-6.1.BETA26\\thaumcraft\\common\\tiles\\devices\\TileMirror.java" "b/bansoukou\\Thaumcraft-1.12.2-6.1.BETA26\\thaumcraft\\common\\tiles\\devices\\TileMirror.java"
index b438e6cd..5b1233cb 100644
--- "a/~bansoukou_unpatched\\Thaumcraft-1.12.2-6.1.BETA26\\thaumcraft\\common\\tiles\\devices\\TileMirror.java"
+++ "b/bansoukou\\Thaumcraft-1.12.2-6.1.BETA26\\thaumcraft\\common\\tiles\\devices\\TileMirror.java"
@@ -311,7 +311,7 @@ ITickable {
             this.instability -= 128;
             this.func_70296_d();
         }
-        if (this.instability > 0 && this.count % 100 == 0) {
+        if (this.instability > 0 && this.count % 5 == 0) {
             --this.instability;
         }
     }
