diff --git "a/~bansoukou_unpatched\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\items\\InitItems.java" "b/bansoukou\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\items\\InitItems.java"
index 9812a150a..586f674ea 100644
--- "a/~bansoukou_unpatched\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\items\\InitItems.java"
+++ "b/bansoukou\\ActuallyAdditions-1.12.2-r152\\de\\ellpeck\\actuallyadditions\\mod\\items\\InitItems.java"
@@ -356,11 +356,11 @@ public final class InitItems {
         itemTeleStaff = new ItemTeleStaff("item_tele_staff");
         itemWingsOfTheBats = new ItemWingsOfTheBats("item_wings_of_the_bats");
         itemDrill = new ItemDrill("item_drill");
-        itemBattery = new ItemBattery("item_battery", 200000, 1000);
-        itemBatteryDouble = new ItemBattery("item_battery_double", 350000, 5000);
-        itemBatteryTriple = new ItemBattery("item_battery_triple", 600000, 10000);
-        itemBatteryQuadruple = new ItemBattery("item_battery_quadruple", 1000000, 30000);
-        itemBatteryQuintuple = new ItemBattery("item_battery_quintuple", 2000000, 100000);
+        itemBattery = new ItemBattery("item_battery", 2000000, 10000);
+        itemBatteryDouble = new ItemBattery("item_battery_double", 3500000, 50000);
+        itemBatteryTriple = new ItemBattery("item_battery_triple", 6000000, 100000);
+        itemBatteryQuadruple = new ItemBattery("item_battery_quadruple", 10000000, 300000);
+        itemBatteryQuintuple = new ItemBattery("item_battery_quintuple", 20000000, 1000000);
         itemDrillUpgradeSpeed = new ItemDrillUpgrade(ItemDrillUpgrade.UpgradeType.SPEED, "item_drill_upgrade_speed");
         itemDrillUpgradeSpeedII = new ItemDrillUpgrade(ItemDrillUpgrade.UpgradeType.SPEED_II, "item_drill_upgrade_speed_ii");
         itemDrillUpgradeSpeedIII = new ItemDrillUpgrade(ItemDrillUpgrade.UpgradeType.SPEED_III, "item_drill_upgrade_speed_iii");
