diff --git "a/~bansoukou_unpatched\\forestry_1.12.2-5.8.2.422\\forestry\\factory\\tiles\\TileRaintank.java" "b/bansoukou\\forestry_1.12.2-5.8.2.422\\forestry\\factory\\tiles\\TileRaintank.java"
index ceb339f4a..ad7811177 100644
--- "a/~bansoukou_unpatched\\forestry_1.12.2-5.8.2.422\\forestry\\factory\\tiles\\TileRaintank.java"
+++ "b/bansoukou\\forestry_1.12.2-5.8.2.422\\forestry\\factory\\tiles\\TileRaintank.java"
@@ -88,7 +88,7 @@ extends TileBase
 implements ISidedInventory,
 ILiquidTankTile {
     private static final FluidStack STACK_WATER = new FluidStack(FluidRegistry.WATER, 1000);
-    private static final FluidStack WATER_PER_UPDATE = new FluidStack(FluidRegistry.WATER, 10);
+    private static final FluidStack WATER_PER_UPDATE = new FluidStack(FluidRegistry.WATER, 30000);
     private final FilteredTank resourceTank;
     private final TankManager tankManager;
     private final ContainerFiller containerFiller;
